DNN Platform/Library/Data/DataProvider.cs

```diff
@@ -3317,14 +3317,14 @@ public virtual IDataReader GetFolderMappingSettings(int folderMappingID)
 
         #region Password History
 
-        public virtual IDataReader GetPasswordHistory(int userId)
+        public virtual IDataReader GetPasswordHistory(int userId, int passwordsRetained, int daysRetained)
         {
-            return ExecuteReader("GetPasswordHistory", GetNull(userId));
+            return ExecuteReader("GetPasswordHistory", GetNull(userId), passwordsRetained, daysRetained);
         }
 
-        public virtual void AddPasswordHistory(int userId, string password, string passwordHistory, int retained)
+        public virtual void AddPasswordHistory(int userId, string password, string passwordHistory, int passwordsRetained, int daysRetained)
```

Comment:
This creates compatibility issues. This must be deprecated and a new method created.